x/ecocredit/server/marketplace/utils.go

```diff
@@ -3,31 +3,21 @@ package marketplace
 import (
 	"context"
 
-	sdk "github.com/cosmos/cosmos-sdk/types"
-
-	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/marketplace/v1"
-	ecoApi "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
-	"github.com/regen-network/regen-ledger/types/math"
 	"github.com/regen-network/regen-ledger/x/ecocredit"
-)
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
 
-// assertHasBalance checks that the account has `qty` credits from the given batch id.
-func assertHasBalance(ctx context.Context, store ecoApi.StateStore, acc sdk.AccAddress, batchId uint64, qty math.Dec) error {
-	res, err := store.BatchBalanceTable().Get(ctx, acc, batchId)
-	if err != nil {
-		return err
-	}
-	tradableBalance, err := math.NewDecFromString(res.Tradable)
-	if err != nil {
-		return err
-	}
-	if tradableBalance.Cmp(qty) == -1 {
-		return ecocredit.ErrInsufficientCredits.Wrapf("cannot create a sell order of %s credits with a balance of %s", qty.String(), res.Tradable)
-	}
-	return nil
-}
+	"github.com/cosmos/cosmos-sdk/types"
+)
 
 // isDenomAllowed checks if the denom is allowed to be used in orders.
-func isDenomAllowed(ctx context.Context, store api.StateStore, denom string) (bool, error) {
-	return store.AllowedDenomTable().Has(ctx, denom)
+func isDenomAllowed(ctx context.Context, denom string, pk ecocredit.ParamKeeper) bool {
+	sdkCtx := types.UnwrapSDKContext(ctx)
+	var params core.Params
+	pk.GetParamSet(sdkCtx, &params)
```

Comment:
I think this is fine but we unwrap sdk context twice where this is used. We could pass sdk context as a param, move these lines outside the helper function, or leave it as is. No strong preference.

Reply:
changed the func parameter to sdk.Context https://github.com/regen-network/regen-ledger/pull/969/commits/8ecf489e45bf6907ca446a256bbc996a0b6fcfb7